### PR TITLE
More web doc updates

### PR DESCRIPTION
**Pull request by rileyjmurray**
Documentation for the following functions was missing from https://www.cvxpy.org/tutorial/functions/index.html:

* conj
* imag
* real
* upper_tri

This PR adds that documentation. It also ...
1. updates that page to reflect how ``kron`` can accept Variables in either argument,
2. explains how signs are determined when applying vector/matrix atoms (the table was incorrect).

### CODE REVIEW

File: doc/source/tutorial/functions/index.rst [hunk 1]
```diff
@@ -503,6 +511,15 @@ scalars, which are promoted.
 
        |decr| for :math:`x \leq 0`
 
+
+   * - :ref:`imag(x) <imag-atom>`
+
+     - imaginary part of a complex number
+     - :math:`x \in \mathbf{C}`
+     - unknown
```
Comment (SteveDiamond) [inline doc/source/tutorial/functions/index.rst:519]: missing |unknown|

Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:519]: Ugh. Jumped the gun.

Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:519]: I'll fix that. Go ahead and point out other issues!

File: doc/source/tutorial/functions/index.rst [hunk 2]
```diff
@@ -820,42 +849,49 @@ and returns a vector or matrix.
...
-:math:`y[k]=\sum_{j=0}^k c[j]x[k-j]`.
+The output :math:`y = \mathbf{conv}(c, x)` has size :math:`n+m-1` and is defined as
+:math:`y_k =\sum_{j=0}^{k} c[j]x[k-j]`.
+
+The output :math:`y = \mathbf{vec}(X)` is the matrix :math:`X` flattened in column-major order into a vector.
+Formally, :math:`y_i = X_{i \bmod{m}, \left \lfloor{i/m}\right \rfloor }`.
 
-The output :math:`x'` of ``vec(X)`` is the matrix :math:`X` flattened in column-major order into a vector.
-Formally, :math:`x'_i = X_{i \bmod{m}, \left \lfloor{i/m}\right \rfloor }`.
+The output :math:`Y = \mathbf{reshape}(X, (m', n'))` is the matrix :math:`X` cast into an :math:`m' \times n'` matrix.
```
Comment (SteveDiamond) [inline doc/source/tutorial/functions/index.rst:889]: Maybe worth mentioning that reshape also supports row-major reordering, with the argument ``order = `C```.

File: doc/source/tutorial/functions/index.rst [hunk 3]
```diff
@@ -820,42 +849,49 @@ and returns a vector or matrix.
...
+The entries are taken from :math:`X` in column-major order and stored in :math:`Y` in column-major order.
+Formally, :math:`Y_{ij} = \mathbf{vec}(X)_{m'j + i}`.
 
-The output :math:`X'` of ``reshape(X, (m', n'))`` is the matrix :math:`X` cast into an :math:`m' \times n'` matrix.
-The entries are taken from :math:`X` in column-major order and stored in :math:`X'` in column-major order.
-Formally, :math:`X'_{ij} = \mathbf{vec}(X)_{m'j + i}`.
+The output :math:`y = \mathbf{upper\_tri}(X)` is formed by concatenating partial rows of :math:`X`.
```
Comment (SteveDiamond) [inline doc/source/tutorial/functions/index.rst:893]: I think upper_tri can include or exclude the diagonal based on a flag.

File: doc/source/tutorial/functions/index.rst (hunk 2 above)
Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:889]: Good point

File: doc/source/tutorial/functions/index.rst [hunk 4]
```diff
@@ -652,6 +669,14 @@ scalars, which are promoted.
      - |convex| convex
      - |incr| incr.
 
+   * - :ref:`real(x) <real-atom>`
+
+     - real part of a complex number
+     - :math:`x \in \mathbf{C}`
+     - unknown
```
Comment (SteveDiamond) [inline doc/source/tutorial/functions/index.rst:676]: missing |unknown|

File: doc/source/tutorial/functions/index.rst (hunk 3 above)
Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:893]: No, it's just the strict upper triangle. There's a helper function in ``upper_tri.py`` called ``vec_to_upper_tri`` which supports a flag, but there isn't an analogous function for going the other way.

Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:893]: That would be a nice feature improvement

File: doc/source/tutorial/functions/index.rst (hunk 4 above)
Comment (rileyjmurray) [reply doc/source/tutorial/functions/index.rst:676]: fixed

**Comment by rileyjmurray** [comment]
@SteveDiamond have you looked over all the changes? If so then I'll commit the fixes directly to release/1.2.x.

**Comment by SteveDiamond** [comment]
Yeah, looks good!

**Pull request merged**